.github/workflows/e2e-tests.yml

```diff
@@ -194,26 +194,26 @@ jobs:
               patch -b data/TraitOne.php < TraitOne.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
-              grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 1 "$OUTPUT"
+              ../bashunit -a contains 'Method TraitsCachingIssue\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' "$OUTPUT"
           - script: |
               cd e2e/trait-caching
               ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
               patch -b data/TraitTwo.php < TraitTwo.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 1 ]
-              grep 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 1 "$OUTPUT"
+              ../bashunit -a contains 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' "$OUTPUT"
           - script: |
               cd e2e/trait-caching
               ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
               patch -b data/TraitOne.php < TraitOne.patch
               patch -b data/TraitTwo.php < TraitTwo.patch
               OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
               echo "$OUTPUT"
-              [ $(echo "$OUTPUT" | wc -l) -eq 2 ]
-              grep 'Method TraitsCachingIssue\\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
-              grep 'Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.' <<< "$OUTPUT"
+              ../bashunit -a line_count 2 "$OUTPUT"
```

Comment:
@Chemaclass thank you.

I have some weird case here.

```
➜  trait-caching git:(bashunit) ✗ echo "$OUTPUT"
/Users/staabm/workspace/phpstan-src/e2e/trait-caching/data/TestClassUsingTrait.php:15:Method TraitsCachingIssue\TestClassUsingTrait::doBar() should return stdClass but returns Exception.
/Users/staabm/workspace/phpstan-src/e2e/trait-caching/data/TestClassUsingTrait.php:29:Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.
➜  trait-caching git:(bashunit) ✗ echo "$OUTPUT" | wc -l | tr -d '[:blank:]'
2
➜  trait-caching git:(bashunit) ✗ ../bashunit -a line_count 2 "$OUTPUT"
✗ Failed: /Users/staabm/workspace/phpstan-src/e2e/trait-caching/data/TestClassUsingTrait.php:29:Method class@anonymous/TestClassUsingTrait.php:20::doBar() should return stdClass but returns Exception.
    Expected '/Users/staabm/workspace/phpstan-src/e2e/trait-caching/data/TestClassUsingTrait.php:15:Method TraitsCachingIssue\TestClassUsingTrait::doBar() should return stdClass but returns Exception.'
    to contain number of lines equal to '2'
```

do you have an idea what is wrong here?

Reply:
I would like to reproduce it locally, so I cloned your fork (`git@github.com:Chemaclass/staabm-phpstan-src.git`), checkout your `bashunit` branch, and I get this error when running composer install - which I think it shouldn't happen as it is a virgin start (I also set php 8.1), so I guess there is a problem on the deps? Did you face the same issue, @staabm ? Any idea how to find a work around to this?

![Screenshot 2024-06-17 at 21 15 14](https://github.com/phpstan/phpstan-src/assets/5256287/5b9fabbb-1df8-4442-bddc-29454deabaae)

Reply:
I think having the actual number would help to understand the issue, 
eg `to contain number of lines equal to '2' but found '1'`
because right now we know it's failing and it's not 2 but what is it? 🤔 I will try to add this now
One guess is that maybe it's only considering one line, but I am not sure, I need to debug and I would like to reproduce this bug in my local. Therefore I wrote the other comment below

Reply:
agree.

Reply:
@staabm this could help to see the actual value https://github.com/TypedDevs/bashunit/pull/263 and enabling adding extra information to the failed assertions in case it might be needed (optional) - feel free to review/comment/approve it if you want (always! 😄)